### DISCUSSION TITLE
Using the backtick key for intellisense trigger of greek symbols

### DISCUSSION THREAD

**Discussion by TryerGit** [James-Yu/LaTeX-Workshop#3623] [Q&A]
Hello,

Following the recommendation [here](https://github.com/James-Yu/LaTeX-Workshop/wiki/Intellisense#-suggestions), I had the following line in my user settings.json file:

"latex-workshop.intellisense.atSuggestion.trigger.latex": "'`'"

The value here is open quotes, single quote backtick single quote close quotes. This does not seem to work after reopening the document. `@` continues to be utilized as the trigger key.

I also tried the value to be: "`" which is open quotes, backtick close quotes. This does not work either.

Pressing backtick in the document immediately renders `|' 
That is, cursor between backtick and single quote and no intellisense of greek symbols turns up even on pressing a in the hope of getting suggestion for \alpha

Use case: the idea of trying to use the backtick key is to try and reduce the number of keystrokes needed -- at present @ requires two keystrokes : Shft + 2, hence my attempt at using the backtick which is a single key press.

Thank you.

**Comment by jlelong** [comment]
Cannot reproduce the issue. Make sure to use the most recent version `9.3.0`. In previous versions, you had to reload VS Code to take into account changes in `latex-workshop.intellisense.atSuggestion.trigger.latex`.

**Reply by TryerGit** [reply]
@jlelong Thank you.

I'd like to understand the difference between 

(1) latex-workshop.intellisense.atSuggestion.trigger.latex, and,

(2) latex-workshop.intellisense.atSuggestionJSON.replace

From my understanding, (2) is used to change what `@a` maps to. So, the following would make `@a` give `\beta`.

`"latex-workshop.intellisense.atSuggestionJSON.replace" : { "@a": "\beta"}`

(1), OTOH, is used to change which other key can perform the role that `@` does.

So, to have backtick in place of @, 

is the setting:

"latex-workshop.intellisense.atSuggestion.trigger.latex": "'`'"
(open quotes single quote backtick single quote close quotes)
[putting in this format because the backtick is used to format as code and hence having to write the above out explicitly]

or

"latex-workshop.intellisense.atSuggestion.trigger.latex": "`" 
(open quotes backtick close quotes)

?

**Reply by jlelong** [reply]
You understanding of (2) is correct.

You have to use "latex-workshop.intellisense.atSuggestion.trigger.latex": "`" but depending on your keyboard layout using the backtick key may not work because it may act as a dead or compose key.

**Reply by TryerGit** [reply]
Hmm...it may well be a versions issue/dead compose key issue but I am unable to get the backtick key to work despite the suggested settings in this thread.

@jlelong 
When you earlier said that you cannot reproduce the issue -- does that mean that you *were* able to get the backtick to work in lieu of @ on your machine? Or that when you pressed the backtick key, VSCode did *not* put your cursor as `|'  (backtick cursor single quote)

Thank you.

**Reply by jlelong** [reply]
When I said at the beginning that I cannot reproduce the issue, I was referring to 
> The value here is open quotes, single quote backtick single quote close quotes. This does not seem to work after reopening the document. `@` continues to be utilized as the trigger key.

You cannot use as a trigger a key which is the opening part of auto closing pair because you cannot avoid the closing part to be inserted.

https://github.com/James-Yu/LaTeX-Workshop/blob/f8a8257f3fcbc02b4e26c255a68bb98b46f6a9f6/syntax/latex-language-configuration.json#L74-L101

**Reply by TryerGit** [reply]
@jlelong 

As a follow up to this -- here is what I tried:

I edited out line 100 which includes the backtick, "`" from the following two files:

`C:\Users\TryerGit\AppData\Local\Packages\CanonicalGroupLimited.Ubuntu20.04onWindows_79rhkp1fndgsc\LocalState\rootfs\home\TryerGit\.vscode-server\extensions\james-yu.latex-workshop-8.21.1\syntax\latex-language-configuration.json
` and
`C:\Users\TryerGit\.vscode\extensions\james-yu.latex-workshop-8.21.1\syntax\latex-language-configuration.json`

and after closing and reopening my document and all instances of VSCode, the backtick key still does not trigger. For e.g., pressing `@a` renders `\alpha`, while `backticka` does not do so.

**Reply by jlelong** [reply]
You seem to be using version `8.21.1` of LaTeX-Workshop, which is quite old. Please update to the latest one `9.3.0`.

**Reply by TryerGit** [reply]
Perhaps a VSCode question. Installing via the extensions tab only gives 8.21.1. Choosing to install a more recent version is not available. Please see attached gif:

![latexworkshop](https://user-images.githubusercontent.com/70434658/211185101-8f733454-936c-458d-b17d-a2c06592a1a0.gif)


Perhaps the last version available depends on the VSCode version itself? My current VSCode version is 

```
Version: 1.60.0 (user setup)
Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
Date: 2021-09-01T10:41:52.311Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19044
```

**Reply by jlelong** [reply]
Your version of VS Code is very old. Please install the latest one `1.74.2`.

**Comment by TryerGit** [comment]
Just to close the loop on this -- the lines that I removed from latex-language-configuration.json so as not to cause a backtick ` to automatically be followed by ' [thereby hindering usage of the backtick as a snippet trigger] were 

```
"surroundingPairs": [
	["`", "'"],	
],

```
in addition to:

```
"autoClosingPairs": [ 
 	["`", "'"] 
 ], 

```

With this change, I am able to not have ` auto surrounded/closed automatically by '

I am then able to use ` as a snippet trigger key for Hypersnips.

Thank you.
